app/[locale]/not-found.tsx

```diff
@@ -1,16 +1,30 @@
-'use client';
-
-import { useTranslations } from 'next-intl';
+import { getLocale } from 'next-intl/server';
 import type { FC } from 'react';
 
-const LocalizedNotFound: FC = () => {
-  const t = useTranslations();
+import { MDXRenderer } from '@/components/mdxRenderer';
+import WithLayout from '@/components/withLayout';
+import { ENABLE_WEBSITE_REDESIGN } from '@/next.constants.mjs';
+import { dynamicRouter } from '@/next.dynamic.mjs';
+
+/** @deprecated replace the legacy page when website redesign is done */
+const notFoundSource = ENABLE_WEBSITE_REDESIGN ? 'new-design/404' : '404';
+
+const LocalizedNotFound: FC = async () => {
+  const locale = await getLocale();
+
+  // Retrieves the markdown source created for the not-found page
+  const { source, filename } = await dynamicRouter.getMarkdownFile(
+    locale,
+    notFoundSource
+  );
+
+  // Parses the source Markdown content and returns a React Component
+  const { MDXContent } = await dynamicRouter.getMDXContent(source, filename);
 
   return (
-    <div className="container">
-      <h2>{t('pages.404.title')}</h2>
-      <h3>{t('pages.404.description')}</h3>
-    </div>
+    <WithLayout layout="centered.hbs">
+      <MDXRenderer Component={MDXContent} />
```

Comment:
I think you could move the glowing div and the main element from the CenteredLayout here.

```tsx
<div className="glowingBackdrop" />

<main>
  <MDXRenderer Component={MDXContent} />
</main>
```

Reply:
Meaning that the CenteredLayout is just:

```tsx
<div className={styles.centeredLayout}>{children}</div>
```

Similar to the Article and Base layout which is what this layout is more intended to be